lib/chef/resource/windows_pagefile.rb

```diff
@@ -98,6 +106,15 @@ class WindowsPagefile < Chef::Resource
               end
             end
           end
+
+          # In the absence of explicitly setting a path, the resource selects the label you used. This can mean a garbage pagefile name.
+          # We test for that here to prevent trying to create a new pagefile when we didnt want to.
+          if pagefile_drive_exist?(pagefile)
+            validate_name
+            create(pagefile) unless exists?(pagefile)
+          elsif !pagefile_drive_exist?(pagefile)
```

Comment:
you shouldn't need to elseif this if you already checked it above

Reply:
done. updated to reflect both these changes.